ray-operator/apis/ray/v1/rayjob_types.go

```diff
@@ -38,10 +38,17 @@ const (
 	JobDeploymentStatusInitializing JobDeploymentStatus = "Initializing"
 	JobDeploymentStatusRunning      JobDeploymentStatus = "Running"
 	JobDeploymentStatusComplete     JobDeploymentStatus = "Complete"
+	JobDeploymentStatusFailed       JobDeploymentStatus = "Failed"
 	JobDeploymentStatusSuspending   JobDeploymentStatus = "Suspending"
 	JobDeploymentStatusSuspended    JobDeploymentStatus = "Suspended"
 )
 
+const (
+	JobReasonSubmissionFailed       string = "SubmissionFailed"
+	JobReasonDeadlineExceeded       string = "DeadlineExceeded"
+	JobReasonApplicationLevelFailed string = "ApplicationLevelFailed"
```

Comment:
```suggestion
	AppFailed string = "AppFailed"
```

Reply:
JobFailed?

Reply:
I feel users might be confused about whether it is a RayJob (KubeRay), a job (Ray), or a Kubernetes submitter job if we use 'JobFailed'.